File: test/token_num.py
import os
import tiktoken as tk

def read_data_from_file(filename):
    """
    Read data from test file
    :param filename: file name
    :return: text_list, model_list, encoding_list
    """
    if not os.path.exists(filename):
        filename = "test.txt"
    with open(filename, 'r', encoding='utf-8') as f:
        lines = f.read().splitlines()
        text_list = lines[0].split(',')
        model_list = lines[1].split(',')
        encoding_list = lines[2].split(',')
    
    return text_list, model_list, encoding_list

def get_token_by_model(text, model):
    """
    Get token count by model
    """
    try:
        encoding = tk.encoding_for_model(model)
        return len(encoding.encode(text))
    except Exception as e:
        print(f"Error model {model}: {e}")
        return 0

def get_token_by_encoding(text, encoding_name):
    """
    Get token count by encoding name
    """
    try:
        encoding = tk.get_encoding(encoding_name)
        return len(encoding.encode(text))
    except Exception as e:
        print(f"Error encoding {encoding_name}: {e}")
        return 0

def test_token_by_model(text_list, model_list):
    for text in text_list:
        for model in model_list:
            num_tokens = get_token_by_model(text, model)
            print(f"text: {text}, model: {model}, token: {num_tokens}")

def test_token_by_encoding(text_list, encoding_list):
    for text in text_list:
        for encoding in encoding_list:
            num_tokens = get_token_by_encoding(text, encoding)
            print(f"text: {text}, encoding: {encoding}, token: {num_tokens}")

if __name__ == '__main__':
    text_list, model_list, encoding_list = read_data_from_file('test/test.txt')
    test_token_by_model(text_list, model_list)
    print("=========================================")
    test_token_by_encoding(text_list, encoding_list)

File: test/token_num.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"

	"github.com/xorvus/tokenizer-go"
)

func main() {
	textList, modelList, encodingList := ReadTestFile()
	testTokenByModel(textList, modelList)
	fmt.Println("=========================================")
	testTokenByEncoding(textList, encodingList)
}

func ReadTestFile() (textList []string, modelList []string, encodingList []string) {
	file, err := os.Open("test.txt")
	if err != nil {
		file, err = os.Open("test/test.txt")
		if err != nil {
			log.Fatal(err)
		}
	}
	defer file.Close()

	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	if err := scanner.Err(); err != nil {
		log.Fatal(err)
	}
	textList = strings.Split(lines[0], ",")
	modelList = strings.Split(lines[1], ",")
	encodingList = strings.Split(lines[2], ",")

	return
}

func getTokenByModel(text string, model string) int {
	tkm, err := tokenizer.ForModel(model)
	if err != nil {
		log.Printf("ForModel error: %v", err)
		return 0
	}

	tokens, err := tkm.EncodeOrdinary(text)
	if err != nil {
		log.Printf("Encode error: %v", err)
		return 0
	}

	return len(tokens)
}

func getTokenByEncoding(text string, encoding string) int {
	tke, err := tokenizer.GetEncoding(tokenizer.Encoding(encoding))
	if err != nil {
		log.Printf("GetEncoding error: %v", err)
		return 0
	}

	tokens, err := tke.EncodeOrdinary(text)
	if err != nil {
		log.Printf("Encode error: %v", err)
		return 0
	}

	return len(tokens)
}

func testTokenByModel(textList []string, modelList []string) {
	for i := 0; i < len(textList); i++ {
		for j := 0; j < len(modelList); j++ {
			fmt.Printf("text: %s, model: %s, token: %d\n", textList[i], modelList[j], getTokenByModel(textList[i], modelList[j]))
		}
	}
}

func testTokenByEncoding(textList []string, encodingList []string) {
	for i := 0; i < len(textList); i++ {
		for j := 0; j < len(encodingList); j++ {
			fmt.Printf("text: %s, encoding: %s, token: %d\n", textList[i], encodingList[j], getTokenByEncoding(textList[i], encodingList[j]))
		}
	}
}
